hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -49,10 +52,10 @@ public NavigablePath(NavigablePath parent, String navigableName) {
 		this( parent, navigableName, null );
 	}
 
-	public NavigablePath(NavigablePath parent, String localName, String alias) {
+	public NavigablePath(NavigablePath parent, String localName, @Nullable String alias) {
 		assert parent != null;
 
-		this.parent = parent;
+		this.parent = NullnessUtil.castNonNull( parent );
```

Comment:
This change shouldn't be necessary.

```suggestion
		this.parent = parent;
```